src/libraries/Microsoft.PowerFx.Interpreter/Functions/ResolvedObjectHelpers.cs

```diff
@@ -20,19 +21,24 @@ public static FormulaValue RecalcFormulaInfo(RecalcFormulaInfo fi)
             return fi._value;
         }
 
-        public static FormulaValue OptionSet(OptionSet optionSet, IRContext irContext)
+        // $$$ - this allows behavior we really don't want ... can we block in the IR?
+        // OptionSets can act like Records: 
+        //   TimeUnit.Hours
```

Comment:
Yeah. We don't enforce this for Option Sets, or legacy enums like TimeUnit/TextPosition/Align. 

For option sets, there's a use case in canvas: The function Choices(OptionSet) results in a list of all the members of the option set, for use in dropdowns and similar.  However, I wonder if we could treat that as a special case as opposed to allowing all these other things as a result of enabling it. 

For legacy enums there's no valid use case, and it seems buggy. We should add a feature flag and block it in binder while we discuss what to do with Canvas.

Reply:
Choices() is a good example - we'll want that in interpreter eventually. But it could be special cased. Ie, Choices impl could get the IExternalOptionSet and operate on that. We don't need to hydrate to a RecordValue.

 So I'll update comment here, but won't make any tangible changes.